src/pages/dynamic-stepper/index.tsx

```diff
@@ -174,9 +174,15 @@ export const DynamicStepper = (): JSX.Element => {
                                     disabled={choice === -1 && index !== steps.length - 1}
                                 >
                                     <StepButton
-                                        onClick={(): void => {
-                                            setActiveStep(index);
-                                        }}
+                                        onClick={
+                                            index === activeStep
+                                                ? (): void => {
+                                                      setActiveStep(-1);
+                                                  }
+                                                : (): void => {
+                                                      setActiveStep(index);
+                                                  }
+                                        }
```

Comment:
You can simplify this a little bit:
```suggestion
                                        onClick={ (): void => {
                                              setActiveStep(index === activeStep ? -1 : index);
                                        } }
```